src/components/File/File.jsx

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright (C) 2018 The Trustees of Indiana University
+SPDX-License-Identifier: BSD-3-Clause
+*/
+import classNames from "classnames";
+import * as PropTypes from "prop-types";
+import * as React from "react";
+import * as Rivet from "../util/Rivet";
+import Icon from "../util/RivetIcons";
+
+const classPrefix = "rvt-file";
+
+function useForceUpdate() {
+  const [value, setValue] = React.useState(0); // integer state
+  return () => setValue((value) => value + 1); // update state to force render
+  // A function that increment 👆🏻 the previous state like here
+  // is better than directly setting `setValue(value + 1)`
+}
+
+const handleFileChange = (onChange, forceUpdate) => (changeEvent) => {
+  if (onChange) {
+    onChange(changeEvent);
+  }
+  forceUpdate();
+};
+
+const File = ({
+  className,
+  id = Rivet.shortuid(),
+  innerRef,
+  label,
+  multiple,
+  onChange,
+  secondary,
+  ...attrs
+}) => {
+  const forceUpdate = useForceUpdate();
+
+  const fileInput = innerRef || React.useRef();
+
+  const form = fileInput.current && fileInput.current.form;
+  if (form) {
```

Comment:
Codecov is suggesting there should be a test for instances when there is no form (you could mock/spy `useForceUpdate` and check that the resulting function is never invoked.

Reply:
Fixing,. Thanks.